### PR TITLE
 YG-296: Added attendance model button and model to the admin grid.

### PR DESCRIPTION
**Pull request by Shivani922**
# Related issues
https://pepcus.atlassian.net/browse/YG-296
https://pepcus.atlassian.net/browse/YG-234

# Screenshots

https://user-images.githubusercontent.com/80804268/228745459-644ed819-a6ad-43ee-a23b-bcf9eb1e81ff.mp4

### CODE REVIEW

File: src/components/routeComponents/memberInformationGrid/SelectedMembersActionWrapper.jsx [hunk 1]
```diff
@@ -149,33 +224,60 @@ class SelectedMembersActionWrapper extends Component {
...
-					/>
+					<StyledButtonRow>
+						<CSVExportButton
+							acl={isCSVExportEnable}
+							selectedMembers={selectedMembers}
+							constants={constants}
+							metaData={metaData}
+						/>
+						<StyledDropdownContainer
+							isEmpty={isEmpty(selectedMembers) ? 'true' : 'false'}>
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/SelectedMembersActionWrapper.jsx:235]: No need to send booleans in strings.
```suggestion
							isEmpty={isEmpty(selectedMembers) ? true : false}>
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+const AttendanceDisplayCard = ({ classroomNo, totalStudent, presentStudent, absentStudent }) => {
+	return (
+		<>
+			<StyledCardContainerWrapper>
+				<StyledCardContainer>
+					<StyledCardHeadingContainer>{classroomNo}</StyledCardHeadingContainer>
+					<StyledCardDivider />
+					<StyledCardInfoContainer>
+						<StyledCardInfo>
+							<StyledCardInfoTitle>Total</StyledCardInfoTitle>
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard.jsx:14]: Can we get these strings like Total, Present etc from JSON constants.

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+	const openModel = () => {
+		setIsModelOpen(true);
+	};
+
+	const closeModal = () => {
+		setIsModelOpen(false);
+	};
+
+	const renderModel = () => {
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx:20]:
```suggestion
	const renderModal = () => {
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import Modal from 'pepcus-core/lib/Modal';
+import Box from 'pepcus-core/lib/Box';
+import ModalHeader from 'components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader';
+import Accordion from 'components/common/Accordion/Accordion';
+import AttendanceDisplayCard from 'components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard';
+
+const AttendanceReportModel = () => {
+	const [isModelOpen, setIsModelOpen] = useState(false);
+
+	const openModel = () => {
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx:12]:
```suggestion
	const openModal = () => {
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import Modal from 'pepcus-core/lib/Modal';
+import Box from 'pepcus-core/lib/Box';
+import ModalHeader from 'components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader';
+import Accordion from 'components/common/Accordion/Accordion';
+import AttendanceDisplayCard from 'components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard';
+
+const AttendanceReportModel = () => {
+	const [isModelOpen, setIsModelOpen] = useState(false);
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx:10]:
```suggestion
	const [isModalOpen, setIsModalOpen] = useState(false);
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+			return (
+				<Modal open={isModelOpen} style={{ padding: '0' }}>
+					<StyledBox
+						backgroundColor="modal"
+						padding="0"
+						borderColor="tertiary"
+						margin="0"
+						width="auto">
+						<ModalHeader onclick={closeModal} />
+						<StyledModelBody>
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx:31]:
```suggestion
						<StyledModalBody>
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+						</StyledModelBody>
+					</StyledBox>
+				</Modal>
+			);
+		}
+		return null;
+	};
+	return (
+		<>
+			<StyledModelLink onClick={openModel}>Attendance Model</StyledModelLink>
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx:44]: Get these strings from JSON constants.
```suggestion
			<StyledModelLink onClick={openModel}>Attendance Report</StyledModelLink>
```

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader.jsx
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader.jsx:1]: There is a typo in directory name.
Please Rename this file and the parent directory to
attendanceReportModal/ModalHeader.jsx

File: src/components/routeComponents/memberInformationGrid/markSelectedMembersAttendance/MarkSelectedMembersAttendance.jsx [hunk 8]
```diff
@@ -163,17 +154,15 @@ class MarkSelectedMembersAttendance extends Component {
 		const { selectedMembers, constants } = this.props;
 		const { MARK_AS_PRESENT } = constants;
 		return (
-			<RowStyled display="inline-block" margin="0 0 0 10px">
-				<ButtonStyled
+			<>
+				<StyledLink
+					isEmpty={isEmpty(selectedMembers) ? 'true' : 'false'}
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/markSelectedMembersAttendance/MarkSelectedMembersAttendance.jsx:159]:
```suggestion
					isEmpty={isEmpty(selectedMembers) ? true : false}
```

File: src/components/common/Accordion/Accordion.jsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import AccordionIcon from 'components/common/Accordion/AccordionIcon';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Accordion = ({ children, day, totalAttendance }) => {
```
[the remark this answers is not in the document]
Comment (Shivani922) [reply src/components/common/Accordion/Accordion.jsx]: Resolve

File: src/components/common/Accordion/Accordion.jsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+	const [isActive, setIsActive] = useState(false);
+	return (
+		<StyledAccordion>
+			<StyledAccordionItem onClick={() => setIsActive(!isActive)}>
+				<StyledAccordionHeader>
+					<StyledAccordionTitle>
+						{day} <span>{totalAttendance}</span>
```
[the remark this answers is not in the document]
Comment (Shivani922) [reply src/components/common/Accordion/Accordion.jsx]: replaced by title and sub title.

File: src/components/routeComponents/memberInformationGrid/SelectedMembersActionWrapper.jsx (hunk 1 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/SelectedMembersActionWrapper.jsx]: resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard.jsx (hunk 2 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceDisplayCard.jsx]: resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx (hunk 3 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx]: resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx (hunk 4 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx]: resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx (hunk 5 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx]: resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx (hunk 6 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx]: Resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx (hunk 7 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/AttendanceReportModel.jsx]: Resolved

File: src/components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader.jsx
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModel/ModelHeader.jsx:1]: Resolved

File: src/components/routeComponents/memberInformationGrid/markSelectedMembersAttendance/MarkSelectedMembersAttendance.jsx (hunk 8 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/markSelectedMembersAttendance/MarkSelectedMembersAttendance.jsx]: Resolved

File: src/components/common/Accordion/Accordion.jsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+import styled from 'styled-components';
+
+const Accordion = ({ children, title, subTitle }) => {
+	const [isActive, setIsActive] = useState(false);
+	return (
+		<StyledAccordion>
+			<StyledAccordionItem onClick={() => setIsActive(!isActive)}>
+				<StyledAccordionHeader>
+					<StyledAccordionTitle>
+						{title} <span>{subTitle}</span>
```
Comment (pepcus-mohit) [inline src/components/common/Accordion/Accordion.jsx:13]: Try to pass the title as a component.

**Review by pepcus-mohit — CHANGES_REQUESTED** [review]

File: src/components/common/Accordion/Accordion.jsx (hunk 11 above)
Comment (Shivani922) [reply src/components/common/Accordion/Accordion.jsx]: added a accordion title component in it.

File: src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+						padding="0"
+						borderColor="tertiary"
+						margin="0"
+						width="auto">
+						<ModalHeader onclick={closeModal} />
+						<StyledModalBody>
+							<Accordion
+								title={
+									<div>
+										Day1 <span>combination attendance</span>
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx:39]: We should get these strings from JSON constants
```suggestion
										Day1 <span>combined attendance</span>
```

**Review by pepcus-mohit — CHANGES_REQUESTED** [review]

File: src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+					<StyledBox
+						backgroundColor="modal"
+						padding="0"
+						borderColor="tertiary"
+						margin="0"
+						width="auto">
+						<ModalHeader onclick={closeModal} />
+						<StyledModalBody>
+							<Accordion
+								title={
```
[the remark this answers is not in the document]
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx]: Implemented the API response

File: src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx (hunk 12 above)
Comment (Shivani922) [reply src/components/routeComponents/memberInformationGrid/attendanceResportModal/AttendanceReportModal.jsx]: Resolved

File: src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx [hunk 14]
```diff
@@ -0,0 +1,396 @@
...
+import Modal from 'pepcus-core/lib/Modal';
+import Box from 'pepcus-core/lib/Box';
+import ModalHeader from 'components/routeComponents/memberInformationGrid/attendanceReportModal/ModalHeader';
+import Accordion from 'components/common/Accordion/Accordion';
+import AttendanceDisplayCard from 'components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceDisplayCard';
+import PropTypes from 'prop-types';
+import { getConstants } from 'reducers/constants';
+import { connect } from 'react-redux';
+
+const AttendanceData = {
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx:12]: The latest changes are available in QA now, Can we remove this mock and replace it with API call.

File: src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx [hunk 15]
```diff
@@ -0,0 +1,396 @@
...
+								return (
+									<Accordion
+										key={methodData}
+										title={
+											<div>
+												{methodData.day}{' '}
+												<span>
+													( {COMBINED_ATTENDANCE}{' '}
+													{methodData.totalPresentStudents} /{' '}
+													{methodData.totalStudents})
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx:326]: Use template literals

File: src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx [hunk 16]
```diff
@@ -0,0 +1,396 @@
...
+			formattedData.push({
+				day: key.toString(),
+				...value,
+				classRooms: classRooms(),
+			});
+		}
+		return formattedData;
+	};
+
+	const AttendanceMethodData = getFormattedData(AttendanceData);
```
Comment (pepcus-mohit) [inline src/components/routeComponents/memberInformationGrid/attendanceReportModal/AttendanceReportModal.jsx:277]:
```suggestion
	const formattedAttendanceResults = getFormattedData(AttendanceData);
```

**Review by pepcus-mohit — CHANGES_REQUESTED** [review]

**Review by pepcus-mohit — APPROVED** [review]

**Review by pepcus-mohit — APPROVED** [review]

**Pull request merged**